### PR TITLE
Add demos and code examples

### PR DESCRIPTION
**Pull request by Blackbaud-SteveBrush**

### CODE REVIEW

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-inline-form?_sp.spa=pr-demos](https://developer.blackbaud.com/skyux-inline-form?_sp.spa=pr-demos)

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-inline-form?_sp.spa=pr-demos](https://developer.blackbaud.com/skyux-inline-form?_sp.spa=pr-demos)

File: src/app/docs/inline-form/inline-form-docs.component.html [hunk 1]
```diff
@@ -154,7 +187,44 @@
...
+  #inlineFormTemplate
+>
+  <div
+    class="sky-form-group"
+  >
+    <label
+      class="sky-control-label"
+      for="my-inline-form"
+    >
+      Page title
```
Comment (blackbaud-johnly) [inline src/app/docs/inline-form/inline-form-docs.component.html]: Should this be `Inline form title`? Seems like that would be more precise.

File: src/app/docs/inline-form/inline-form-docs.component.ts [hunk 2]
```diff
@@ -1,11 +1,64 @@
...
+  public demoSettings: {
+    inlineFormConfig: SkyInlineFormConfig;
+    showForm: boolean;
+    title: string;
+  } = {
+    inlineFormConfig: {
+      buttonLayout: SkyInlineFormButtonLayout.SaveCancel
+    },
+    showForm: false,
+    title: 'My inline form title'
```
Comment (blackbaud-johnly) [inline src/app/docs/inline-form/inline-form-docs.component.ts]: Superminor, but I'd delete `My` here and in the other instances and just go with `Inline form title`. (Not sure if we want a more real-world example given the bare bones nature of the demo.)

Also, this title shows up initially when I see the page, and then it shows up again as placeholder text in the field after clicking the initial link. It doesn't really make sense in the field, and it's not italicized like placeholder text is supposed to be. Can we either without placeholder text or display the placeholder text with the `placeholder` attribute on the input like we demo in the form classes docs?

File: src/app/docs/inline-form/inline-form-docs.component.ts [hunk 3]
```diff
@@ -1,11 +1,64 @@
...
   selector: 'app-inline-form-docs',
-  templateUrl: './inline-form-docs.component.html',
-  changeDetection: ChangeDetectionStrategy.OnPush
+  templateUrl: './inline-form-docs.component.html'
 })
-export class InlineFormDocsComponent { }
+export class InlineFormDocsComponent {
+
+  public inlineFormButtonLayoutChoices: SkyDocsDemoControlPanelRadioChoice[] = [
+    { label: 'DoneCancel', value: SkyInlineFormButtonLayout.DoneCancel },
```
Comment (blackbaud-johnly) [inline src/app/docs/inline-form/inline-form-docs.component.ts:24]: I can see where it's implemented, but the demo I see uses Done and Cancel buttons. Is that a typical use case? It seems to me that given that there is an input field, Save and cancel buttons make more sense here and are probably the more typical use case.

**Review by blackbaud-johnly — COMMENTED** [review]
A few comments/suggestions...

File: src/app/docs/inline-form/inline-form-docs.component.ts (hunk 3 above)
Comment (Blackbaud-SteveBrush) [reply src/app/docs/inline-form/inline-form-docs.component.ts:24]: Good catch. Done.

File: src/app/docs/inline-form/inline-form-docs.component.ts (hunk 2 above)
Comment (Blackbaud-SteveBrush) [reply src/app/docs/inline-form/inline-form-docs.component.ts]: I reworked the demo to hopefully make more sense.

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-inline-form?_sp.spa=pr-demos](https://developer.blackbaud.com/skyux-inline-form?_sp.spa=pr-demos)

**Review by blackbaud-johnly — DISMISSED** [review]

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-inline-form?_sp.spa=pr-demos](https://developer.blackbaud.com/skyux-inline-form?_sp.spa=pr-demos)

File: src/app/public/plugin-resources/code-examples/inline-form/basic/inline-form-demo.component.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    firstName?: string;
+  } = { };
+
+  public inlineFormConfig: SkyInlineFormConfig = {
+    buttonLayout: SkyInlineFormButtonLayout.SaveCancel
+  };
+
+  public showForm: boolean = false;
+
+  public firstName: string = 'Jane';
```
Comment (Blackbaud-AlexKingman) [inline src/app/public/plugin-resources/code-examples/inline-form/basic/inline-form-demo.component.ts]: Check your alphabetical ordering here. Looks like the other example needs a fix too.

File: src/app/public/plugin-resources/code-examples/inline-form/repeaters/inline-form-demo.component.html [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+  <sky-repeater-item *ngFor="let item of items"
+    [inlineFormConfig]="inlineFormConfig"
+    [inlineFormTemplate]="inlineFormTemplate"
+    [showInlineForm]="activeInlineFormId === item.id"
+    (inlineFormClose)="onInlineFormClose($event)"
+  >
+
+    <sky-repeater-item-title>
+      <div
+        class="pull-left"
```
Comment (Blackbaud-AlexKingman) [inline src/app/public/plugin-resources/code-examples/inline-form/repeaters/inline-form-demo.component.html]: These `pull-` classes are using CSS floats... I don't think we want to suggest that our consumers use them. Could we show this example without using floats?

File: src/app/docs/inline-form/inline-form-docs.component.html [hunk 6]
```diff
@@ -8,7 +8,48 @@
   </sky-docs-demo-page-summary>
 
   <sky-docs-demo>
-    See documentation for how to setup demos.
+    <sky-docs-demo-control-panel
+      (selectionChange)="onDemoSelectionChange($event)"
+    >
+      <sky-docs-demo-control-panel-section>
+        <sky-docs-demo-control-panel-radio-group
+          heading="Default button configurations"
```
Comment (Blackbaud-AlexKingman) [inline src/app/docs/inline-form/inline-form-docs.component.html]: Should this just say `Button configurations` ? What does the word "default" give us here?

**Review by Blackbaud-AlexKingman — COMMENTED** [review]
Looks great. Just a couple things.

File: src/app/docs/inline-form/inline-form-docs.component.html (hunk 6 above)
Comment (Blackbaud-SteveBrush) [reply src/app/docs/inline-form/inline-form-docs.component.html]: I didn't include the `Custom` type and didn't want to use the word "prefabs". I can go with "Button configurations" if you're okay with excluding "Custom".

File: src/app/public/plugin-resources/code-examples/inline-form/repeaters/inline-form-demo.component.html (hunk 5 above)
Comment (Blackbaud-SteveBrush) [reply src/app/public/plugin-resources/code-examples/inline-form/repeaters/inline-form-demo.component.html]: These are hold-overs from skyux2 (and the current code example). I'll replace them with something else.

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-inline-form?_sp.spa=pr-demos](https://developer.blackbaud.com/skyux-inline-form?_sp.spa=pr-demos)

**Review by Blackbaud-AlexKingman — APPROVED** [review]

**Pull request merged**